src/main/java/teammates/logic/core/FeedbackSessionsLogic.java

```diff
@@ -1110,6 +1110,29 @@ private StringBuilder getRowsOfPossibleRecipientsInCsvFormat(
         }
         return sessionsToSendEmailsFor;
     }
+    
+    /**
+     * Criteria: must be published, publishEmail must be enabled and
+     * resultsVisibleTime must be custom.
+     * 
+     * @return returns a list of sessions that require automated emails to be
+     *         sent as they are published
+     */
+    public List<FeedbackSessionAttributes> getFeedbackSessionsWhichNeedAutomatedUnpublishedEmailsToBeSent() {
+        List<FeedbackSessionAttributes> sessions =
+                fsDb.getFeedbackSessionsWithUnsentUnpublishedEmail();
+        List<FeedbackSessionAttributes> sessionsToSendEmailsFor =
+                new ArrayList<FeedbackSessionAttributes>();
+
+        for (FeedbackSessionAttributes session : sessions) {
+            // automated emails are required only for custom unpublish times
+            if (!session.isPublished() && session.isPublishedEmailEnabled()
```

Comment:
`List<FeedbackSessionAttributes> sessions =   fsDb.getFeedbackSessionsWithUnsentUnpublishedEmail();`
This code saves us from that issue. We get only those sessions which are unpublished and Emails are not sent, and for the new session we have `sentUnpublishedEmail = true` . So we won't get any newly created session.